index.html

```diff
@@ -9222,6 +9225,256 @@ <h3 id="example-2">
+                <li>Set <var>change-rate</var> to the difference between <var>
+                  attenuation</var> and <var>detector average</var>. Multiply
+                  <var>change-rate</var> by the <var>envelope rate</var>. Add
+                  <var>change-rate</var> to <var>detector average</var>, and
+                  clamp <var>detector average</var> to a maximum of 1.0.
+                </li>
+                <li>If attacking, set <var>attack</var> to <var>detector
+                average</var> minus <var>compressor gain</var>. Set
+                <var>compressor gain</var> to the multiplication of
+                <var>attack</var> by <var>envelope rate</var>. Clamp
+                <var>compressor gain</var> to a maximum of 1.0.
+                </li>
+                <li>Else, multiply <var>compressor gain</var> by <var>envelope
+                rate</var>
+                </li>
+                <li>Compute <var>final gain</var> to be <var>compressor
+                gain</var> multiplied by the return value of <a>computing the
+                makeup gain</a>.
+                </li>
+                <li>Compute <var>metering gain</var> to be <var>final
+                gain</var>, <a href="#linear-to-decibel">converted to
+                decibel</a>.
+                </li>
+                <li>
+                  <a href="atomic">Atomically</a> set the internal slot
+                  [[\reduction]] to the value of <var>metering gain</var>.
+                </li>
+              </ol>
+            </li>
+          </ol>
+          <p>
+            The makeup gain is a fixed gain stage that only depends on ratio,
+            knee and threshold parameter of the compressor, and not on the
+            input signal. The intent here is to increase the output level of
+            the compressor so it is comparable to the input level.
+          </p>
+          <p>
+            <dfn>Computing the makeup gain</dfn> means executing the following
+            steps:
+          </p>
+          <ol>
+            <li>Let <var>full range gain</var> be the value returned by
+            applying the <a>compression curve</a> to the value 1.0.
+            </li>
+            <li>Let <var>full range makeup gain</var> the inverse of <var>full
+            range gain</var>.
+            </li>
+            <li>Return the result of taking 0.6 power of <var>full range makeup
+            gain</var>.
+            </li>
+          </ol>
+          <p>
+            <dfn id="#envelope-rate">Computing the envelope rate</dfn> is done
+            by applying a function to the ratio of the <var>compressor
+            gain</var> and the <var>detector average</var>. User-agents are
+            allowed to choose the shape of the attack and release curve.
+            However, they MUST respect the following constraints:
+          </p>
+          <ul>
+            <li>The input to the curve MUST be the ratio of the <var>compressor
```

Comment:
Rather than the "input to the curve", this is more clearly described as the envelope rate itself (which in turn determines the curve). So... "the envelope rate MUST be calculated from the ratio of..."